pytket/phir/phirgen.py

```diff
@@ -169,58 +169,82 @@ def classical_op(exp: LogicExp, *, bitwise: bool = False) -> JsonDict:
+                "angles": angles,
+                "args": [[arg_to_bit(cmd.qubits[0]), arg_to_bit(cmd.qubits[1])]],
+            }
+        case _:  # single-qubit gates
+            qop = {
+                "qop": gate,
+                "angles": angles,
+                "args": [arg_to_bit(cmd.qubits[0])],
+            }
+    return qop
+
+
 def convert_subcmd(op: tk.Op, cmd: tk.Command) -> JsonDict | None:
-    """Return PHIR dict give op and its arguments."""
+    """Return PHIR dict given a tket op and its arguments."""
     if op.is_gate():
-        try:
-            gate = tket_gate_to_phir[op.type]
-        except KeyError:
-            if op.type == tk.OpType.Phase:
-                # ignore global phase
-                return {"mop": "Skip"}
-            logging.exception("Gate %s unsupported by PHIR", op.get_name())
-            raise
-        angles = (op.params, "pi") if op.params else None
-        qop: JsonDict
-        match gate:
-            case "Measure":
-                qop = {
-                    "qop": gate,
-                    "returns": [arg_to_bit(cmd.bits[0])],
-                    "args": [arg_to_bit(cmd.args[0])],
-                }
-            case ("CX"
-                | "CY"
-                | "CZ"
-                | "RXX"
-                | "RYY"
-                | "RZZ"
-                | "R2XXYYZZ"
-                | "SXX"
-                | "SXXdg"
-                | "SYY"
-                | "SYYdg"
-                | "SZZ"
-                | "SZZdg"
-                | "SWAP"
-            ):  # two-qubit gates  # fmt: skip
-                qop = {
-                    "qop": gate,
-                    "angles": angles,
-                    "args": [[arg_to_bit(cmd.qubits[0]), arg_to_bit(cmd.qubits[1])]],
-                }
-            case _:  # single-qubit gates
-                qop = {
-                    "qop": gate,
-                    "angles": angles,
-                    "args": [arg_to_bit(cmd.qubits[0])],
-                }
-        return qop
+        return convert_gate(op, cmd)
 
     out: JsonDict | None = None
     match op:  # non-quantum op
         case tk.BarrierOp():
-            out = {"meta": "barrier", "args": [arg_to_bit(qbit) for qbit in cmd.qubits]}
+            if op.data:
+                # See https://github.com/CQCL/tket/blob/0ec603986821d994caa3a0fb9c4640e5bc6c0a24/pytket/pytket/qasm/qasm.py#L419-L459
+                match op.data[0:5]:
+                    case "sleep":
+                        dur = op.data.removeprefix("sleep(").removesuffix(")")
+                        out = {
+                            "mop": "Idle",
+                            "args": [arg_to_bit(qbit) for qbit in cmd.qubits],
+                            "duration": (dur, "s"),
+                        }
+                    case "order" | "group":
+                        raise NotImplementedError
+                    case _:
+                        raise TypeError
```

Comment:
Maybe it would be nice to add a message explaining the error, including the `op.data` string that was not recognized? Similarly for the `NotImplementedError`.